README.md

```diff
@@ -2,15 +2,75 @@
 
 ## Intro
 
-This repository contains the backend logic for the dfusion exchange based on the specification, see [here](github.com/gnosis/dex-research)
+This repository contains the backend logic for the dfusion exchange based on [this specification](github.com/gnosis/dex-research).
 
-## Architecture
+It contains two sub-projects that both implement the market mechanism described above in different ways. An fully on-chain solution with instant finality but limited scalability (referred to as "StableX") and a preliminary version that intends to achieves scalability by offloading computation and data-storage off-chain using an [optimistic roll-up](https://medium.com/plasma-group/ethereum-smart-contracts-in-l2-optimistic-rollup-2c1cef2ec537) approach. The latter is in early development stage and not yet ready for use.
 
-<p align="center">
-  <img src="documentation/architecture.png" alt="dex-services architecture" width="500">
-</p>
+## Getting Started
+
+### Requirements
+
+- Rust (stable)
+- NodeJS <=11.0, starting with version 12 some deprecated APIs were removed that cause `scrypt`, `keccak`, `secp256k1`, and `sha3` packages to fail to build
+- Docker and Docker-compose (stable)
+
+The project may work with other versions of these tools but they are not tested.
+
+### Installation
+
+Clone the repository, its submodule, and run the container
+```bash
+git clone git@github.com:gnosis/dex-services.git
+cd dex-services
+git submodule update --init
+docker-compose up -d ganache-cli
+(cd dex-contracts && yarn && npx truffle compile && npm run networks-inject && npx truffle migrate)
+```
+
+## StableX
+
+The StableX system only consists of a simple service that queries the relevant auction information (orders and balances) directly from the blockchain. It then tries to find and submit a valid solution as soon as the order collection phase for a given auction ends.
+
+The repo ships with a very naive solver, that can at the moment only match two orders between the fee token (*token0*) and another token if those orders overlap. A more sophisticated solver using a linear programming approach is not open sourced at the moment.
+
+### Running StableX
+
+```bash
+docker-compose up stablex
+```
+
+In order to setup the environment and make a first deposit/order you can run:
+
+```
+cd dex-contracts
+npx truffle exec scripts/stablex/setup_environment.js
```

Comment:
I thinks it's more clear if you explain a bit what this lines do using a comment. i.e:

Actually, I'm not sure someone would realize this is for running a fake thing in a local ganache with fake data.

```bash
# Basic Test Setup: Lists 3 fake tokens, and creates a deposit for 3 fakes users
npx truffle exec scripts/stablex/setup_environment.js
```